### PR TITLE
Add ability to edit an existing access code

### LINKED ISSUE #302: AccessCodeForm - Add ability to edit an existing code
An extension of https://github.com/seamapi/react/issues/236, current implementation only has the ability to add a new code. This task is to implement the edit features for existing codes.

#### `<AccessCodeTable />`:
- [x] Add 'edit' menu

#### `<AccessCodeDetails/>`

- [x] Add 'Edit code' button

#### `<AccessCodeForm />`

- [ ] ~~Add `onEdit` prop~~ Not implemented, as handling edit inside the form instead:

```tsx
  if (accessCodeId != null) {
    return (
      <AccessCodeEditForm
        accessCodeId={accessCodeId}
        onBack={onBack}
        className={className}
      />
    )
  }

  if (deviceId != null) {
    return (
      <AccessCodeAddForm
        deviceId={deviceId}
        onBack={onBack}
        className={className}
      />
    )
```

### PR DESCRIPTION
**Pull request by mikewuu**
Closes #302 

### Notable things

- `/update` isn't currently handled in `fake-seam-connect`, which means storybook is broken.

https://github.com/seamapi/react/assets/11449462/93e3991d-d654-4103-a993-71f535a7139c

### CODE REVIEW

File: src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx
Comment (razor-x) [inline src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx:72]: I think we can simply things by passing in `submit` and `isLoading` (or maybe `isSubmitting`), but it makes the types a bit harder to work out. 🤔

**Review by razor-x — CHANGES_REQUESTED** [review]
Let's wait until access code update is supported by the fake.

Nice work so far splitting this up though!

File: src/lib/seam/access-codes/use-update-access-code.ts
Comment (mikewuu) [reply src/lib/seam/access-codes/use-update-access-code.ts]: Updated in https://github.com/seamapi/react/pull/312/commits

File: src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx]: Updated in 082990a657d04dcc252a897444377ab6304eac5c

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:154]: Yep, will handle it in a follow-up issue: https://github.com/seamapi/react/issues/303

File: src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: Yea, I've moved things around in https://github.com/seamapi/react/pull/312/commits/140a430f4424af96a7663ef3af2bc6edf841c9cb. Instead of having the generic `<AccessCodeForm/>` be at the top-level, users should use the more specific components:

- `<CreateAccessCodeForm/>`
- `<EditAccessCodeForm/>`

If they want to handle the submission themselves, they can of course use the more low-level `<AccessCodeForm/>`.

The above structure has made all the complexity of types, and cross-cutting hooks disappear. It does deviate a bit from the spec though.

File: src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx]: Yep, updated in https://github.com/seamapi/react/commit/140a430f4424af96a7663ef3af2bc6edf841c9cb for the `<AccessCodeForm/>` to just take an `onSubmit`. Much cleaner now IMO.

File: src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx
Comment (razor-x) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: Nice. I am thinking we should just have `CreateAccessCodeForm` and `EditAccessCodeForm` and not export `AccessCodeForm`. I know that was not the original thought, but it's a cleaner api IMO. Same reason we have `/access_code/update` and `/access_code/create`.

Comment (mikewuu) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: We could do. Should I rename / add the stories for this?

Comment (razor-x) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: We should confirm with @dawnho first. I strongly recommend this approach.

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx
Comment (razor-x) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:154]: Ok. We do need to disable it via the hard coded const before merging this though, just like we have for `disableCreateAccessCode` right now.

File: src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx
Comment (razor-x) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: If we go this route, I guess we should move `AccessCodeForm` to the ui folder and have `CreateAccessCodeForm` and `EditAccessCodeForm` as their own folders / elements just like the other main components.

Comment (razor-x) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: @mikewuu All good to go ahead with the plan.

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:156]: Added in https://github.com/seamapi/react/pull/312/commits/580b8c2fba1d27d45931496403df79e0b73cd6cd

Comment (mikewuu) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:156]: I've set it to `true` because it should be fine to edit the name / schedule (start/end times)

File: src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx
Comment (mikewuu) [reply src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx]: Refactored in https://github.com/seamapi/react/pull/312/commits/bebc0a36ef6a1b5fc6710c51d4c5e30db704d54d

**Comment by mikewuu** [comment]
@razor-x this should be good to merge, the fake is working as expected.

https://github.com/seamapi/react/assets/11449462/119fa936-5a93-4a67-9d5b-0dc8b10cffe1

File: src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx
Comment (razor-x) [inline src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx:22]: I know this looks redundant, but talking with @dawnho we realized there will be a common set of "global" config like this, so we want to keep the prop name the same for all the components that use them.

```suggestion
  disableAccessCodeEdit?: boolean
```

File: src/lib/seam/components/CreateAccessCodeForm/CreateAccessCodeForm.tsx
Comment (razor-x) [inline src/lib/seam/components/CreateAccessCodeForm/CreateAccessCodeForm.tsx:51]:
```suggestion
function useSubmitCreateAccessCode(onSuccess?: () => void): {
```

File: src/lib/seam/components/EditAccessCodeForm/EditAccessCodeForm.tsx
Comment (razor-x) [inline src/lib/seam/components/EditAccessCodeForm/EditAccessCodeForm.tsx:66]:
```suggestion
function useSubmitEditAccessCode(
```

File: src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx
Comment (razor-x) [reply src/lib/seam/components/AccessCodeDetails/AccessCodeDetails.tsx:22]: However, we can't add this prop in this PR, it must be hard coded false as in the other component since we do not want to anything to use the access code form until it's generally released.

**Review by razor-x — APPROVED** [review]
Just a few minor typos / naming suggestions. Approving so this can be merged, however we do need to disable the edit view until release.

**Pull request merged**
